portfolio/src/main/webapp/script.js

```diff
@@ -16,13 +16,81 @@
  * Adds a random greeting to the page.
  */
 function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
-
+  const greetings = [
+    "Hello world!",
+    "¡Hola Mundo!",
+    "你好，世界！",
+    "Bonjour le monde!",
+  ];
   // Pick a random greeting.
   const greeting = greetings[Math.floor(Math.random() * greetings.length)];
-
   // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
+  const greetingContainer = document.getElementById("greeting-container");
   greetingContainer.innerText = greeting;
 }
+
+function addDate() {
+  const date = new Date();
+  const dateContainer = document.getElementById("date-container");
+  dateContainer.innerText = date;
+}
+
+function openSidebar() {
+  document.getElementById("mySidebar").style.display = "block";
+}
+
+function closeSidebar() {
+  document.getElementById("mySidebar").style.display = "none";
+}
+
+function displayImage() {
+  var x = document.getElementById("fav_animal");
+  if (x.style.display === "none") {
+    x.style.display = "block";
+  } else {
+    x.style.display = "none";
+  }
+}
+
+function hideImage() {
+  var x = document.getElementById("fav_animal");
+  x.style.display = "block";
+}
+
+/** Adds text form DataServlet to the webpage using arrow */
+function getText() {
+  fetch("/data")
+    .then((response) => response.text())
+    .then((text) => {
+      document.getElementById("places-lived-container").innerText = text;
+    });
+}
+
+/**fetching the JSON arraylist string from server */
+function fetchList() {
+  fetch('/data').then(response => response.json()).then((placesLived) => {
+    console.log(placesLived);
+    const placesLivedElement = document.getElementById('places-lived-container');
+    placesLivedElement.innerHTML = '';
+    placesLivedElement.appendChild(
```

Comment:
Can you just do this in a for loop instead of hard coding it? That way, if you add more places, it gets updated automatically.

Reply:
done